POST /party/join 서비스 및 컨트롤러 구현

deepredk opened the pull request:
* PartyMemberService에 join 메서드 추가 구현
* PartyService getAnyNotFullParty private 메서드 추가구현
* PartyJoinResult 만듬
* ottsharing.serviceFee 커스텀 프로퍼티 만듬

----

* MoneyService쪽 실수 슬쩍 수정
* 경고 해결하려고 database-platform 속성을 넣는 삽질을 했으나 오류가 발생해서 다시 제거함

closes #54 
closes #58

deepredk committed:
feat: ottsharing.serviceFee 커스텀 프로퍼티 만듬

deepredk committed:
feat: POST /party/join 서비스 및 컨트롤러 구현

* PartyMemberService에 join 메서드 추가 구현
* PartyService getAnyNotFullParty private 메서드  추가구현
* PartyJoinResult 만듬

deepredk committed:
fix: MoneyService 리팩토링 및 application.yml database-platform 속성 제거

저 속성이 오히려 오류를 발생시킴

sue4869 commented on src/main/resources/META-INF/spring-configuration-metadata.json:
```diff
@@ -0,0 +1,8 @@
+      "type": "java.lang.Integer"
+    }
+  ]
+}
```
이 파일은 혹시 어디에 쓰이는 건가용? application.yml에 serviceFee를 넣어주기 위해 쓰이는 건가용?

sue4869 commented on src/main/resources/META-INF/spring-configuration-metadata.json:
```diff
@@ -0,0 +1,8 @@
+    {
+      "name": "ottsharing.serviceFee",
+      "type": "java.lang.Integer"
+    }
```
@deepredk
이 파일은 혹시 어디에 쓰이는 건가용? application.yml에 serviceFee를 넣어주기 위해 쓰이는 건가용?

deepredk replied:
@sue4869 네네 맞습니다
이 파일이 없으면 application.yml에서 Warning이 떠요

sue4869 replied:
오홍 알려주셔서 감사합니당

sue4869 commented on src/main/java/kr/flab/ottsharing/service/PartyService.java:
```diff
@@ -124,16 +133,28 @@ public MyParty fetchMyParty(User user) {
-        // User Repository 구조 개편으로 코드 정상적으로 동작하지 않음
-        /*
+    @Transactional
+    public PartyJoinResult join(User user) {
```
@deepredk 여기서 @Transactional을 쓰시는 건 어떤 걸 방지하기 위함인가용? 쓰신 의도가 궁금합니당

deepredk replied:
@sue4869 지불만 되고 가입은 안 될 경우를 생각해서 Transactional을 사용했어요!

sue4869 commented on src/main/java/kr/flab/ottsharing/service/PartyService.java:
```diff
@@ -124,16 +133,28 @@ public MyParty fetchMyParty(User user) {
+    public PartyJoinResult join(User user) {
+        if (moneyService.pay(user, serviceFee) == PayResult.NOT_ENOUGH_MONEY) {
+            return PartyJoinResult.NOT_ENOUGH_MONEY; 
+        } 
```
moneyService.pay(user,serviceFee)에 대한 결과를 담는 객체를 따로 만드는 것과 같이 어떠한 조치가 있었으면 좋겠어요.  현재 이 코드으로만 볼때는 moneyService.pay(user,serviceFee)를 통해 돈이 충분한지 안한지만 판단하는 역할만 수행하는 것 같은 착각이 일어나요. 실제로 moneyService.pay(user,serviceFee)가 어떻게 구현되었는지 알기 위해 해당 서비스로 들어가야지만 그제서야 지불 과정도 있다는 것이 파악 된다는 문제점이 있어서요

deepredk replied:
@sue4869 
오 좋은 지적해주신것 같습니다 반영할게요!

deepredk replied:
@sue4869 반영했습니다! 확인 부탁드려요

sue4869 replied:
네 확인했습니당 고생많으셨어요

deepredk committed:
refactor: pay()의 반환값을 따로 변수에 담도록하여 지불결과라는 의미를 명확히 함

deepredk committed:
(충돌해결) Merge branch 'develop' into feature/54

# Conflicts:
#	src/main/java/kr/flab/ottsharing/controller/PartyController.java
#	src/main/java/kr/flab/ottsharing/service/PartyMemberService.java
#	src/main/java/kr/flab/ottsharing/service/PartyService.java
